verification/verify.go

```diff
@@ -127,18 +130,17 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		// Verification passed successfully, so use this set of params
 		return params, nil
 	}
-	sv.count++
```

Comment:
Curious about this choice - why shouldn't a fatal error count towards the number of retries? This would result in taking more retries than the policy calls for. Under pathological conditions (eg, perhaps there is a bug in how we count pixels), we could be retrying until we run out of orchestrators.

Reply:
@yondonfu made this decision, perhaps he will be able to explain this better than I would.

Reply:
I restored the count field and previous handling such that fatal errors will count towards the retries count.

Your TLDR says `ErrAudioMismatch` but the same holds for `ErrPixelMismatch`, no?  I believe the PR in it's current state is correct, if not please clarify.

Reply:
> I restored the count field and previous handling such that fatal errors will count towards the retries count.

👍 

> Your TLDR says ErrAudioMismatch but the same holds for ErrPixelMismatch, no?

See this section of my previous comment:

> However, I now realize that results that trigger ErrPixelMismatch could also be valid - the # of reported pixels might be wrong, but the video/audio content could be fine. The only error that should make a result ineligible to be appended to sv.results is ErrAudioMismatch because if this error is triggered then we know with certainty that the audio was tampered so this result should not be eligible for insertion into a playlist.

I think the following update to the conditional evaluated prior to appending a result to `sv.results` should work:

```
if err != ErrAudioMismatch && IsRetryable(err) {
    ...
    sv.results = append(sv.results, r)
}
```

Reply:
If we make that change, then we'll no longer be using the `IsFatal()` function nor considering the `Fatal` error type at all.   Should I go ahead and revert it all?

Reply:
Additionally, making that change causes the test that @j0sh asked to be added to fail.  So, this PR no longer fixes the bug he referred to.  Is there something else we need to do for that?   I've pushed up the changes in the failing state just so that we can be on the same page.

Reply:
Done.  Also had to change @j0sh 's test to use `ErrAudioMismatch` instead of `ErrPixelMismatch`

Reply:
We do still ultimately want to distinguish between fatal vs. non-fatal retryable errors because we'll want to re-add Os back to the working set for non-fatal retryable errors after finishing a segment - right now we just [remove Os for all retryable errors](https://github.com/livepeer/go-livepeer/blob/bbb72c0f54a9f70ed65611b74abcc52add85bead/server/broadcast.go#L534). As already mentioned, we can handle re-adding Os back to the working set for non-fatal retryable errors outside of this PR though. 

I do see that given the latest changes that the distinction between fatal vs. non-fatal retryable errors wouldn't actually be useful until we do the above. I'm ok with leaving that out of this PR since the relevant code has already been removed.